packages/webapp/src/pages/Integrations/providerConfigKey/Endpoints/components/ScriptSettings.tsx

```diff
@@ -0,0 +1,308 @@
+            type: flow.type!,
+            scriptName: flow.name
+        });
+        if ('error' in res.json) {
+            toast({ title: 'An unexpected error occurred', variant: 'error' });
+        } else {
+            toast({ title: `Upgraded successfully`, variant: 'success' });
+            await mutate((key) => typeof key === 'string' && key.startsWith('/api/v1/integrations'));
+            setOpenFrequency(false);
+        }
+
+        setLoadingUpgrade(false);
+    };
+
+    // Frequency
+    const [openFrequency, setOpenFrequency] = useState(false);
+    const [loadingFrequency, setLoadingFrequency] = useState(false);
+    const [frequency, setFrequency] = useState(flow.runs);
+    const [frequencyError, setFrequencyError] = useState<string | null>(null);
+
+    const onEditFrequency = async () => {
+        setLoadingFrequency(true);
+
+        // Already deployed, we just need to enable
+        const res = await apiFlowUpdateFrequency(
+            env,
+            { id: flow.id! },
+            {
+                provider: integration.integration.provider,
+                providerConfigKey: integration.integration.unique_key,
+                type: flow.type!,
+                scriptName: flow.name,
+                frequency
+            }
+        );
+        if ('error' in res.json) {
+            toast({ title: 'An unexpected error occurred', variant: 'error' });
+        } else {
+            toast({ title: `Update successfully`, variant: 'success' });
+            await mutate((key) => typeof key === 'string' && key.startsWith('/api/v1/integrations'));
+            setOpenFrequency(false);
+        }
+
+        setLoadingFrequency(false);
+    };
+
+    const onFrequencyChange = (value: string) => {
+        setFrequency(value);
+
+        const reg = value.match(frequencyRegex);
+        if (!reg || !reg.groups) {
+            setFrequencyError('Format should be "every (number) (minutes|hours|days)", e.g: "every 1 hour", "every 20 days"r');
+            return;
+        }
+
+        const unit = reg.groups['unit'];
+        const amount = parseInt(reg.groups['amount'], 10);
+
+        if (unit.startsWith('m') && (amount < 5 || !amount)) {
+            setFrequencyError('The minimum frequency is 5 minutes');
+            return;
+        }
+        setFrequencyError(null);
+    };
+
+    const isSync = flow.type === 'sync';
+
+    return (
+        <Drawer
+            direction="right"
+            snapPoints={open ? [drawerWidth] : []}
+            handleOnly={true}
+            noBodyStyles={true}
+            dismissible={true}
+            disablePreventScroll={true}
+            modal={true}
+            open={open}
+            // onClose={() => setOpen(false)}
+            onOpenChange={setOpen}
+        >
+            <DrawerTrigger asChild>
+                <Button variant={'zombie'} className="text-text-light-gray">
+                    <GearIcon />
+                    Settings
```

Comment:
UI feedback. The gear icon  + settings label made me think that I would be able to change things in the UI but the panel is read-only without any mention of how to change those values. Not a big deal but I wonder if I am the only one we had this first reaction when seeing this button

Reply:
You can edit the "Enabled" setting so I guess in a way that makes sense. I would have expected some kind of change on hover so that I know that this button + icon does something. Right now there is nothing on hover.

Reply:
It is also very confusing that the same word "Settings" is used for the sync but also for the integration Settings. I think this should be changed for sure.

![image](https://github.com/user-attachments/assets/b79880fe-0a03-46b6-84cb-ddb58a6d087a)

Reply:
Also with this box, which settings are we telling the user to go to?

![image](https://github.com/user-attachments/assets/c1122910-74f7-48a1-9eb2-70f0258b1471)

Reply:
cc @bastienbeurier

Reply:
I agree with all points. 

What about: 
- Changing "Settings" to "Script configuration" 
- The button changes on hover
- The warning box is changed to "..., go to the script configuration."
- In the Script Configuration, we can have a short introduction paragraph (similar to the Figma design), saying: 

> Nango endpoints are powered by integration scripts. Some of the following configurations can only be changed by modifying this underlying script. If the source is a template, you will need to extend it to change certain configurations ([guide](https://docs.nango.dev/customize/guides/extend-an-integration-template)).